packages/connectivity/lib/connectivity.dart

```diff
@@ -60,11 +81,24 @@ class Connectivity {
   /// make a network request. It only gives you the radio status.
   ///
   /// Instead listen for connectivity changes via [onConnectivityChanged] stream.
-  Future<ConnectivityResult> checkConnectivity() async {
+  Future<ConnectivityResult> checkConnectivity(
+      {bool checkSubtype = false}) async {
     final String result = await methodChannel.invokeMethod<String>('check');
     return _parseConnectivityResult(result);
   }
 
+  /// Checks the network mobile connection subtype of the device.
+  /// Return EDGE for 2G, HSDPA for 3G and LTE for 4G depending on the connection of the mobile connection
+  /// if it is connected.
+  ///
+  /// Return none if there is no connections
+  ///
+  /// Return unknown if it is connected but there is not connection subtype info. eg. Wifi
+  Future<ConnectionSubtype> getNetworkSubtype() async {
```

Comment:
more doc to point users to the [checkConnectivity] for who wants to check the main type.

Reply:
i added some docs. Please review and let me know if they are up to standard.